determined-ai/determined
# chore: remove admin flag check in delete_model & delete_model_version [DET-9596]

## Description
There are currently hard checks on the admin flag when deleting model registries/versions; instead we'd like these authentication checks to rely on the authz.

Additionally the logic for deleting model registries and versions was split up to allow for greater granularity of control.




## Test Plan
- Create a 2 model registries (at least 1 with multiple versions) for each an admin & non-admin account 
- Ensure that the admin can: delete their own model registries & versions & delete other users' models/versions
- Ensure that non-admin's can: delete their own model registries & version & CANNOT delete other users' models/versions




## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
 [DET-9596]






[DET-9596]: https://hpe-aiatscale.atlassian.net/browse/DET-9596?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
